src/Transports.AspNetCore/WebSockets/BaseSubscriptionServer.cs

```diff
@@ -0,0 +1,488 @@
+                }
+                finally
+                {
+                    disposer?.Dispose();
+                }
+            }
+            else if (result.Executed)
+            {
+                await SendSingleResultAsync(message, result);
+            }
+            else
+            {
+                await SendErrorResultAsync(message, result);
+            }
+        }
+        catch (OperationCanceledException) when (CancellationToken.IsCancellationRequested)
+        {
+            throw;
+        }
+        catch (ExecutionError error)
+        {
+            if (!Subscriptions.Contains(messageId, dummyDisposer))
+                return;
+            await SendErrorResultAsync(message, error);
+        }
+        catch (Exception ex)
+        {
+            if (!Subscriptions.Contains(messageId, dummyDisposer))
+                return;
+            var error = await HandleErrorDuringSubscribeAsync(message, ex);
+            await SendErrorResultAsync(message, error);
+        }
+    }
+
+    /// <summary>
+    /// Creates an <see cref="ExecutionError"/> for an unknown <see cref="Exception"/>.
+    /// </summary>
+    protected virtual Task<ExecutionError> HandleErrorDuringSubscribeAsync(OperationMessage message, Exception ex)
+        => Task.FromResult<ExecutionError>(new UnhandledError("Unable to set up subscription for the requested field.", ex));
+
+    /// <summary>
+    /// Sends a single result to the client for a subscription or request, along with a notice
+    /// that it was the last result in the event stream.
+    /// </summary>
+    protected virtual async Task SendSingleResultAsync(OperationMessage message, ExecutionResult result)
+    {
+        await SendDataAsync(message.Id!, result);
+        await SendCompletedAsync(message.Id!);
+    }
+
+    /// <summary>
+    /// Sends an execution error to the client during set-up of a subscription.
+    /// </summary>
+    protected virtual Task SendErrorResultAsync(OperationMessage message, ExecutionError executionError)
+        => SendErrorResultAsync(message.Id!, executionError);
+
+    /// <summary>
+    /// Sends an execution error to the client during set-up of a subscription.
+    /// </summary>
+    protected virtual Task SendErrorResultAsync(string id, ExecutionError executionError)
+        => SendErrorResultAsync(id, new ExecutionResult { Errors = new ExecutionErrors { executionError } });
+
+    /// <summary>
+    /// Sends an error result to the client during set-up of a subscription.
+    /// </summary>
+    protected virtual Task SendErrorResultAsync(OperationMessage message, ExecutionResult result)
+        => SendErrorResultAsync(message.Id!, result);
+
+    /// <summary>
+    /// Sends an error result to the client during set-up of a subscription.
+    /// </summary>
+    protected abstract Task SendErrorResultAsync(string id, ExecutionResult result);
+
+    /// <summary>
+    /// Sends a data packet to the client for a subscription event.
```

Comment:
```suggestion
    /// Sends a data packet to the client for a GraphQL request (typically subscription event).
```

I would check all xml comments to focus on the fact that any requests can be performed and not just subscriptions.